SampleMerchant/public/index.html

```diff
@@ -106,11 +148,142 @@ <h1>Sample Merchant</h1>
                 log.e(e.message);
               })
           }
+          
+          request.addEventListener('shippingoptionchange', function(e) {
+            e.updateWith(new Promise(function(resolve) {
+              resolve(updateDetailsForOptionChange(details, request.shippingOption));
+            }));
+          });
+
+          request.addEventListener('shippingaddresschange', function(e) {
+            e.updateWith(new Promise(function(resolve) {
+                resolve(updateDetailsForAddressChange(details, request.shippingAddress));
+            }));
+          });
+
+          request.addEventListener('paymentmethodchange', function(e) {
+            e.updateWith(new Promise(function(resolve) {
+                resolve(updateDetailsForPaymentMethodChange(details, e));
+            }));
+          });
+
           return request;
         }
 
         let request = buildPaymentRequest();
 
+        /**
+         * Adjusts the total price based on the currently selected shipping option, 
+         * the destination country's currency, and applied promotion code.
+         * @param  {string} price - The original price tag.
+         * @return {string} The adjusted price tag.
+         */
+        function adjustTotalPrice(price) {
+          let adjustedPrice = parseFloat(price);
+          // Apply shipping cost.
+          if (details.shippingOptions[0].selected)
+            adjustedPrice += 5
+          else
+            adjustedPrice += 25
+
+          // Apply promotion discount.
+          if (promotionApplied)
+            adjustedPrice -= 5
+
+          if (destinationCountry === 'CA')
+            return (adjustedPrice * 1.2).toFixed(2)
+          else
+            return adjustedPrice;
```

Comment:
Missing `.toFixed(2)` here?

Without it, it will return a `number` instead of a `string`.

Reply:
Good catch, thanks

Reply:
done